spacepy/pycdf/istp.py

```diff
@@ -492,6 +492,15 @@ def _validhelper(cls, v, rng=True):
                     continue
                 if firstdim: #Add pseudo-record dim
                     attrval = numpy.reshape(attrval, (1, -1))
+            if (v.attrs.type(which) in spacepy.pycdf.lib.timetypes) != (v.type() in spacepy.pycdf.lib.timetypes):
```

Comment:
We're also using v.attrs.type(which) and v.type() a lot, starting back on line 472. Should we just assign to a variable at the point where we decide to execute the loop? (i.e. first line after the `continue`). Call 'em attrtype and vtype or something?